dopamine/replay_memory/circular_replay_buffer.py

```diff
@@ -178,6 +179,8 @@ def __init__(self,
         [math.pow(self._gamma, n) for n in range(update_horizon)],
         dtype=np.float32)
 
+    lock_lib.initialize_lock(self)
```

Comment:
do you want to guard this with a constructor param? maybe you could have a param `lock_attribute_name` which defaults to `lock_lib._DEFAULT_LOCK_ATTR_NAME`, and which disables locking if set to `None`.

alternatively, consider having a constructor param `lock` that defaults to `threading.Lock()`, and is passed in to the call to `lock_lib.initialize_lock()`. this has the benefit that it will also allow you to inject the mock lock more easily in the tests.

Reply:
I opted for adding a `lock` attribute defaulting to `threading.Lock()`.

Having a `lock_attribute_name` attribute would add a bit of complexity as the `lock_attribute_name` would also need to be specified to the method decorators.